src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -98,184 +98,341 @@ def filter_devices(args):
     unused_devices = [device for device in args.devices if not device.used_by_ceph]
     # only data devices, journals can be reused
     used_devices = [device.abspath for device in args.devices if device.used_by_ceph]
-    args.filtered_devices = {}
+    filtered_devices = {}
     if used_devices:
         for device in used_devices:
-            args.filtered_devices[device] = {"reasons": ["Used by ceph as a data device already"]}
+            filtered_devices[device] = {"reasons": ["Used by ceph as a data device already"]}
         logger.info("Ignoring devices already used by ceph: %s" % ", ".join(used_devices))
     if len(unused_devices) == 1:
         last_device = unused_devices[0]
         if not last_device.rotational and last_device.is_lvm_member:
             reason = "Used by ceph as a %s already and there are no devices left for data/block" % (
                 last_device.lvs[0].tags.get("ceph.type"),
             )
-            args.filtered_devices[last_device.abspath] = {"reasons": [reason]}
+            filtered_devices[last_device.abspath] = {"reasons": [reason]}
             logger.info(reason + ": %s" % last_device.abspath)
             unused_devices = []
 
-    return unused_devices
+    return unused_devices, filtered_devices
 
 
 class Batch(object):
 
     help = 'Automatically size devices for multi-OSD provisioning with minimal interaction'
 
     _help = dedent("""
-    Automatically size devices ready for OSD provisioning based on default strategies.
+    Automatically size devices ready for OSD provisioning.
+    The 'auto' subcommand will automtatically determine the strategy (based on
```

Comment:
I don't understand why another (3rd level) sub command is needed

Reply:
As mentioned above

> one cannot have an optional positional argument (the subcommands) and a mandatory positional argument list (the existing DEVICES arguments)

Reply:
This is certainly not needed unless you are objecting to anything besides a 3rd level sub-command implementation. We already have this type of arrangement for zap (positional  arguments for devices as well as flags defining devices)

Reply:
As mentioned in other places this RFC is mostly about the argparsing. I'm absolutely not married to the subcommand structure, I just think this is the clearest approach.

My worry about adding a bunch of new flags is that `batch`'s argument list will grow rather long and confusing. I'll prepare a commit with this style of argparsing to compare.